### PR TITLE
readme: Improve instructions for running Spark

### PR DESCRIPTION
**Pull request by kayousterhout**
This commit adds more background information about why to use
Kelda to run Spark, and adds more information in a FAQ.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,5 +150,46 @@ looks like:
...
-### More information
+### Use the Spark UI
+
+The Kelda Spark blueprint opens all of the ports necessary to access the various
+Spark UIs:
+* **Application (Per-Job) UI** The UI for a particular application is available
+on port 4040 on the `spark-driver` machine. This is typically the most useful UI
+for debugging performance of the tasks in a particular workload. The full URL
+of the application UI is printed when starting a Spark shell (as in the example
+UI above), and it's also shown in the `PUBLIC IP` column of the kelda show
```
Comment (kklin) [inline README.md]: Maybe quote "kelda show" in backticks?

File: README.md [hunk 2]
```diff
@@ -139,5 +150,46 @@ looks like:
...
+Spark's standalone scheduler to schedule applications on worker machines. The
+standalone scheduler runs on the `spark-master` machine, and runs a UI on port
+8080 of that machine. As with the other UIs, the full address of this UI is
+shown in the `PUBLIC IP` column of the `kelda show` container output. This
+UI shows all of the workers in the Spark cluster.
+
+For more information about the various Spark UIs, refer to the Spark
+documentation.
+
+## More information
```
Comment (kklin) [inline README.md]: Should this be beneath the FAQ?

Comment (kayousterhout) [reply README.md]: I wanted to put it above in case the FAQ gets longer (since people may not scroll to the bottom of it)

**Comment by kayousterhout** [comment]
@kklin thanks for looking at this -- it's good to go now!

**Pull request merged**